scss/mixins/_icon-mixins.scss

```diff
@@ -0,0 +1,15 @@
+@function str-replace($string, $search, $replace: '') {
+    $index: str-index($string, $search);
+
+    @if $index {
+        @return str-slice($string, 1, $index - 1) + $replace + str-replace(str-slice($string, $index + str-length($search)), $search, $replace);
+    }
+
+    @return $string;
+}
+
+@function svg-fill ($svg, $color) {
+  $fillColor: str-replace(#{$color}, '#', '%23');
+  @return str-replace($svg, '%3Csvg', '%3Csvg%20%20fill%3D%27#{$fillColor}%27');
```

Comment:
So I understand, you're replacing the fill color in a string that directly represents the svg?